### DISCUSSION TITLE
Help with rules (it seems that the rule is in loop)

### DISCUSSION THREAD

**Discussion by walberjunior** [emqx/emqx#14270] [Q&A]
Hello everyone.
I've been testing EMQX for a while now and I'm trying to create a rule using republish, but the rule seems to be running in a loop.
And another problem is that even using the topic zigbee2mqtt/Varanda/center any message sent to zigbee2mqtt/Varanda/* executes the rule.

```
SELECT
  payload
FROM
  "zigbee2mqtt/Varanda/center"
WHERE
  payload = '{"state":"ON"}' OR payload = '{"state":"OFF"}'

```

![image](https://github.com/user-attachments/assets/2f6393ca-9214-43a5-a1c9-ff389873f635)

![image](https://github.com/user-attachments/assets/b72c14f0-714a-4d46-8e43-14a39393ee24)

**Comment by zmstone** [comment]
starting from 5.8, there is a `direct_dispatch` option for republish actions, set it to true to avoid loop.
ref: https://github.com/emqx/emqx/pull/13516

You can also consider EMQX Enterprise edition to make use of the Schema Validation feature where you can set a expression to validate if the payload is valid, invalid messages can be configured to be dropped.
